rita_client/src/dashboard/devices_on_lan.rs

```diff
@@ -0,0 +1,366 @@
+                                        found_match = true;
+                                    } else {
+                                        break;
+                                    }
+                                }
+                                Err(_) => break,
+                            }
+                        }
+                        if index == 3 && found_match {
+                            if entry.to_string().len() <= 3 {
+                                return mac_addr.to_string();
+                            } else {
+                                return entry.to_string();
+                            }
+                        }
+                    }
+                }
+                mac_addr.to_string()
+            } else {
+                mac_addr.to_string()
+            }
+        }
+        Err(_e) => mac_addr.to_string(),
+    }
+}
+
+/// This is an endpoint to grab all the lan devices mapping to ip address and
+/// returns the json request populated with the related hardware information
+pub async fn get_devices_lan_endpoint(_req: HttpRequest) -> HttpResponse {
+    let command_response = grab_ip_neigh();
+    match command_response {
+        Ok(output) => {
+            // Extra checking since output struct is ambigious on how it works
+            if !output.stdout.is_empty() {
+                let arp_table = parse_ip_neigh(String::from_utf8_lossy(&output.stdout).to_string());
+                info!("Arp table length: {:?}", arp_table);
+
+                let lan_arp_table = prune_non_lan_entries(arp_table);
+                let consolidated_lan_arp_table = consolidate_wlan_arp_table(lan_arp_table);
+
+                let rita_client = settings::get_rita_client();
+                let hardware_to_check = match get_hardware_info(rita_client.network.device) {
+                    Ok(info) => info,
+                    Err(e) => {
+                        return HttpResponse::InternalServerError()
+                .json(format!("Unable to grab rita client hardware information. Failed with error {:?}",e));
+                    }
+                };
+
+                HttpResponse::Ok().json(generate_lan_device(
+                    consolidated_lan_arp_table,
+                    hardware_to_check,
+                ))
+            } else {
+                HttpResponse::InternalServerError().finish()
+            }
+        }
+        Err(e) => HttpResponse::InternalServerError().json(e.to_string()),
+    }
+}
+
+/// Checks if 2 ip addresses does not fall within each others subnets
+/// This is done by bitmasking each address and comparing to see if they are the same
+/// OUTPUTS:
+/// Returns false if the ip addresses do not match the same format ipv4 with ipv6 or ipv6 with ipv4
+/// Returns true if the addresses do not fall within each other subnets
+fn check_ip_in_subnet(address1: IpAddr, address2: IpAddr, subnet_mask: u8) -> bool {
```

Comment:
why exactly is this needed? To see if target ip's fall within the lan subnet? Please note that